File: OnlineJudge/LeetCode/codes/9.java
import java.io.FileInputStream;
import java.io.FileNotFoundException;
import java.io.FileOutputStream;
import java.io.PrintStream;
import java.util.Scanner;

class Solution {
  public boolean isPalindrome(int x) {
    if (x < 0) {
      return false;
    }
    int res = x;
    int num = 0;
    int r = Integer.MAX_VALUE / 10;
    while (x > 0) {
      if (num > r) {
        return false;
      } else if (num == r && (x % 10) > 7) {
        return false;
      }
      num = num * 10 + x % 10;
      x /= 10;
    }

    return res == num;
  }

  public static void main(String[] args) throws FileNotFoundException {
    FileInputStream fin = new FileInputStream("./assets/fipt.txt");
    PrintStream fot = new PrintStream(new FileOutputStream("./assets/fopt.txt"));

    System.setIn(fin);
    System.setOut(fot);

    Solution sol = new Solution();

    sol.input();
  }

  public void input() {
    Scanner in = new Scanner(System.in);
    while (in.hasNext()) {
      n = in.nextInt();
      for (int i = 0; i < n; i++) {
        a = in.nextInt();
        solve(a);
      }
    }

    in.close();
  }

  public void solve(int x) {
    boolean res;
    res = isPalindrome(x);
    otput(res);

  }

  public void otput(Boolean x) {
    System.out.println(x);
  }

  private int n;
  private int a;
}

File: OnlineJudge/LeetCode/codes/9.cpp
#include <math.h>

#include <algorithm>
#include <cstdio>
#include <cstdlib>
#include <cstring>
#include <iostream>
#include <map>
#include <queue>
#include <stack>
#include <string>
#include <vector>

using namespace std;

class Solution {
 public:
  bool isPalindrome(int x) {
    if (x < 0) {
      return false;
    }
    int res = x;
    int num = 0;
    int r = INT_MAX / 10;
    while (x) {
      if (num > r) {
        return false;
      } else if (num == r && (x % 10) > 7) {
        return false;
      }
      num = num * 10 + x % 10;
      x /= 10;
    }

    return res == num;
  }

  void input(void) {
    while (~scanf("%d", &n)) {
      for (int i = 0; i < n; i++) {
        cin >> a;
        solve(a);
      }
    }
  }

  void solve(int x) {
    bool res;
    res = isPalindrome(x);
    otput(res);
  }

  void otput(bool x) { cout << x << endl; }

 private:
  int n;
  int a;
};

int main() {
  freopen("./assets/fipt.txt", "r", stdin);
  freopen("./assets/fopt.txt", "w", stdout);

  Solution sol;

  sol.input();

  return 0;
}
